### PR TITLE
Update "Working with objects"

### PR DESCRIPTION
**Pull request by alattalatta**
#### Summary

Fixed and updated the doc here and there.

#### Motivation

- Code blocks don't comply MDN code formatting rules, such as missing semi and things.
- Code uses `var`, changed to `let` and `const`.
  - ~~Didn't use `const` to avoid having to explain here why a const object properties can be changed... Which I think can be confusing. 🤔~~
- A few sentences are out of its place, such as line 156. (upstream)
- Some links are outdated.

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/guide/working_with_objects/index.md [hunk 1]
```diff
@@ -97,87 +97,82 @@ You can use the bracket notation with [`for...in`](/en-US/docs/Web/JavaScript/Re
...
-
 ## Creating new objects
 
 JavaScript has a number of predefined objects. In addition, you can create your own objects. You can create an object using an [object initializer](/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer). Alternatively, you can first create a constructor function and then instantiate an object invoking that function in conjunction with the `new` operator.
 
 ### Using object initializers
 
-In addition to creating objects using a constructor function, you can create objects using an [object initializer](/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer). Using object initializers is sometimes referred to as creating objects with literal notation. "Object initializer" is consistent with the terminology used by C++.
+Using object initializers is sometimes referred to as creating objects with literal notation. "Object initializer" is consistent with the terminology used by C++.
```
Comment (wbamberg) [inline files/en-us/web/javascript/guide/working_with_objects/index.md:149]: It seems a shame to lose that link. Maybe:

```suggestion
Using [object initializers](/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer) is sometimes referred to as creating objects with literal notation. "Object initializer" is consistent with the terminology used by C++.
```

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks @alattalatta , this is a good improvement!

I had one very minor comment. I would merge this PR even without it being addressed but please accept it if you agree. On this:

> * Didn't use `const` to avoid having to explain here why a const object properties can be changed... Which I think can be confusing. 🤔

I agree that `const` can be confusing, mostly because it is badly taught. For that reason I think it's important to use it consistently in our docs and explain it clearly from the start.

We discussed this issue recently: https://github.com/mdn/content/discussions/10389 and agreed to use `const` in situations like this. 

But I'd merge this PR anyway because it's much better than what was there before.

File: files/en-us/web/javascript/guide/working_with_objects/index.md (hunk 1 above)
Comment (alattalatta) [reply files/en-us/web/javascript/guide/working_with_objects/index.md:149]: Ahh, I thought this was too repeating with

> You can create an object using an [object initializer](/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer).

just above. I'll have it restored with `const`s.

Comment (wbamberg) [reply files/en-us/web/javascript/guide/working_with_objects/index.md:149]: Oh sorry! Now you point that out, I agree.

Comment (alattalatta) [reply files/en-us/web/javascript/guide/working_with_objects/index.md:149]: Alright, hehe, just the `const`s then.

**Review by wbamberg — APPROVED** [review]
Very nice cleanup. Thank you @alattalatta !

**Pull request merged**